pipeline-scripts/commonlib.groovy

```diff
@@ -236,17 +244,23 @@ def shell(arg) {
     def script = arg.script  // keep a copy of original script
     def truncScript = (script.size() <= 75) ? script : script[0..35] + "..." + script[-36..-1]
 
-    // ensure a clean space to save output files
-    if (!shellIndex++) { sh("rm -rf shell") }
-
-    def filename = "shell/sh.${shellIndex}." + truncScript.replaceAll( /[^\w-.]+/ , "_").take(80)
-    sh("mkdir -p ${filename}")
-    filename += "/sh.${shellIndex}"
+    // put shell result files in a subdir specific to this shell call
+    def threadShellIndex = getNextShellCounter()
+    def filename = "shell/sh.${threadShellIndex}." + truncScript.replaceAll( /[^\w-.]+/ , "_").take(80)
```

Comment:
If `shell(...)` is used inside of a `dir(){}` block, I think it will tend to drop shellDir's around the filesystem. Making these absolute paths with `env.WORKSPACE` might be worthwhile.

Reply:
Seemed worthwhile to me. Turned into a bigger project than expected but hopefully the outcome is actually clearer.